docroot/profiles/custom/cgov_site/modules/custom/cgov_event/src/Controller/ICalendarController.php

```diff
@@ -76,23 +90,29 @@ public static function create(ContainerInterface $container) {
    */
   public function download($nid) {
     // Load event node.
-    $node = $this->entityTypeManager()->getStorage('node')->load($nid);
+    /** @var \Drupal\node\NodeInterface $storage */
+    $storage = $this->entityTypeManager()->getStorage('node');
+    $node = $storage->load($nid);
     if (empty($node)) {
       return FALSE;
     }
 
     $start_date = NULL;
     $end_date = NULL;
 
+
     if ($node->hasField('field_event_start_date') && !empty($node->get('field_event_start_date')->first())) {
       // Convert the date to the Timezone of the user requesting.
-      $start_date = $node->get('field_event_start_date')->date->format(
+      $start_date_obj = new DrupalDateTime($node->get('field_event_start_date')->getString(), DateTimeItemInterface::STORAGE_TIMEZONE);
```

Comment:
Can you get the date out and just cast it to `\DateTime`? Something like:
```
/** @var \DateTime **/
$date = $node->get('field_event_start_date')?->first()->get('date');
$start_date = $date->format('Y-m-d H:i:s');
```

Reply:
Resolved.